pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -74,52 +79,69 @@ func (t ResolvedPipelineRunTask) IsDone(facts *PipelineRunFacts) bool {
 	return t.Skip(facts) || t.IsSuccessful() || t.IsFailure()
 }
 
+// IsCustomTask returns true if the PipelineTask references a Custom Task.
+func (t ResolvedPipelineRunTask) IsCustomTask() bool {
+	return t.CustomTask
+}
+
 // IsSuccessful returns true only if the taskrun itself has completed successfully
 func (t ResolvedPipelineRunTask) IsSuccessful() bool {
-	if t.TaskRun == nil {
-		return false
+	if !t.IsCustomTask() {
```

Comment:
Can we switch the condition from `!t.IsCustomTask()` to `t.IsCustomTask()` wherever possible, for example:

```	
if t.IsCustomTask() {
		return t.Run != nil && t.Run.IsSuccessful()
	}
	if t.TaskRun == nil {
		return false
	}
	return t.TaskRun.IsSuccessful()
```

Reply:
Agreed.

Reply:
Done.